infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -0,0 +1,218 @@
+    this.sshManager = sshManager;
+  }
+
+  @Override
+  @Traced
+  public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+
+    TracingTags.WORKSPACE_ID.set(identity::getWorkspaceId);
+
+    StringBuilder sshConfig = new StringBuilder();
+    Map<String, String> map = new HashMap<>();
+
+    try {
+      List<SshPairImpl> sshPairs = sshManager.getPairs(identity.getOwnerId(), "vcs");
+      if (!sshPairs.isEmpty()) {
+        for (SshPair sshPair : sshPairs) {
+          if (isNullOrEmpty(sshPair.getName()) || isNullOrEmpty(sshPair.getPrivateKey())) {
+            continue;
+          }
+          Secret secret =
+              new SecretBuilder()
+                  .addToData(
+                      SSH_PRIVATE_KEY,
+                      Base64.getEncoder().encodeToString(sshPair.getPrivateKey().getBytes()))
+                  .withType(SECRET_TYPE_SSH)
+                  .withNewMetadata()
+                  .withName(getValidNameForSecret(sshPair.getName()))
+                  .endMetadata()
+                  .build();
+
+          k8sEnv.getSecrets().put(secret.getMetadata().getName(), secret);
+
+          k8sEnv
+              .getPodsData()
+              .values()
+              .forEach(p -> addSshKeySecret(secret.getMetadata().getName(), p.getSpec()));
+
+          sshConfig.append(buildConfig(sshPair.getName()));
+        }
+
+        map.put(SSH_CONFIG, sshConfig.toString());
+        ConfigMap configMap =
+            new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName("sshconfigmap")
+                .endMetadata()
+                .withData(map)
+                .build();
+
+        k8sEnv.getConfigMaps().put(configMap.getMetadata().getName(), configMap);
+        k8sEnv.getPodsData().values().forEach(p -> addConfigFile(p.getSpec()));
+      }
+    } catch (ServerException e) {
+      LOG.warn("Unable get SSH Keys ", e);
+    }
+  }
+
+  private void addSshKeySecret(String secretName, PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName(secretName)
+                .withSecret(new SecretVolumeSourceBuilder().withSecretName(secretName).build())
+                .build());
+    List<Container> containers = podSpec.getContainers();
+    containers.forEach(
+        container -> {
+          VolumeMount volumeMount =
+              new VolumeMountBuilder()
+                  .withName(secretName)
+                  .withNewReadOnly(false)
+                  .withReadOnly(false)
+                  .withMountPath(SSH_BASE_CONFIG_PATH + secretName)
+                  .build();
+          container.getVolumeMounts().add(volumeMount);
+        });
+  }
+
+  private void addConfigFile(PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName("configvolume")
```

Comment:
I think `configvolume` is too abstract name for volume that may cause a collision with user-defined volumes.